From abbotti@mev.co.uk  Fri Nov 20 09:12:32 2009
From: Ian Abbott <abbotti@mev.co.uk>
Date: Fri, 20 Nov 2009 11:32:38 +0000
Subject: Staging: comedi: pcl816: update convert_src mask for AI cmdtest
Cc: Frank Mori Hess <fmhess@users.sourceforge.net>, Ian Abbott <abbotti@mev.co.uk>, Greg Kroah-Hartman <gregkh@suse.de>
Message-ID: <1258716758-9033-4-git-send-email-abbotti@mev.co.uk>


The COMEDI_CMDTEST ioctl needs to clear unsupported bits in the
struct comedi_cmd's convert_src and other *_src members.  This
needs fixing in the pcl816 driver's AI cmdtest.

Signed-off-by: Ian Abbott <abbotti@mev.co.uk>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/staging/comedi/drivers/pcl816.c |    4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

--- a/drivers/staging/comedi/drivers/pcl816.c
+++ b/drivers/staging/comedi/drivers/pcl816.c
@@ -488,7 +488,9 @@ static int pcl816_ai_cmdtest(struct come
 	if (!cmd->scan_begin_src || tmp != cmd->scan_begin_src)
 		err++;
 
-	if (!(cmd->convert_src & (TRIG_EXT | TRIG_TIMER)))
+	tmp = cmd->convert_src;
+	cmd->convert_src &= TRIG_EXT | TRIG_TIMER;
+	if (!cmd->convert_src || tmp != cmd->convert_src)
 		err++;
 
 	tmp = cmd->scan_end_src;
